spring-boot-starter-camunda/src/main/java/io/camunda/zeebe/spring/client/configuration/ZeebeClientConfiguration.java

```diff
@@ -0,0 +1,198 @@
+
+public class ZeebeClientConfiguration implements io.camunda.zeebe.client.ZeebeClientConfiguration {
+
+  @Autowired
+  private ZeebeClientConfigurationProperties properties;
+
+  @Autowired
+  private Authentication authentication;
+
+  @Lazy // Must be lazy, otherwise we get circular dependencies on beans
+  @Autowired
+  private JsonMapper jsonMapper;
+
+  @Lazy
+  @Autowired(required = false)
+  private List<ClientInterceptor> interceptors;
+
+  @Lazy
+  @Autowired
+  private ZeebeClientExecutorService zeebeClientExecutorService;
+
+  @PostConstruct
+  public void applyLegacy() {
+    // make sure environment variables and other legacy config options are taken into account (duplicate, also done by  qPostConstruct, whatever)
+    properties.applyOverrides();
+  }
+
+  @Override
+  public String getGatewayAddress() {
+    return properties.getGatewayAddress();
+  }
+
+  @Override
+  public String getDefaultTenantId() {
+    return properties.getDefaultTenantId();
+  }
+
+  @Override
+  public List<String> getDefaultJobWorkerTenantIds() {
+    return properties.getDefaultJobWorkerTenantIds();
+  }
+
+  @Override
+  public int getNumJobWorkerExecutionThreads() {
+    return properties.getNumJobWorkerExecutionThreads();
+  }
+
+  @Override
+  public int getDefaultJobWorkerMaxJobsActive() {
+    return properties.getDefaultJobWorkerMaxJobsActive();
+  }
+
+  @Override
+  public String getDefaultJobWorkerName() {
+    return properties.getDefaultJobWorkerName();
+  }
+
+  @Override
+  public Duration getDefaultJobTimeout() {
+    return properties.getDefaultJobTimeout();
+  }
+
+  @Override
+  public Duration getDefaultJobPollInterval() {
+    return properties.getDefaultJobPollInterval();
+  }
+
+  @Override
+  public Duration getDefaultMessageTimeToLive() {
+    return properties.getDefaultMessageTimeToLive();
+  }
+
+  @Override
+  public Duration getDefaultRequestTimeout() {
+    return properties.getDefaultRequestTimeout();
+  }
+
+  @Override
+  public boolean isPlaintextConnectionEnabled() {
+    return properties.isPlaintextConnectionEnabled();
+  }
+
+  @Override
+  public String getCaCertificatePath() {
+    return properties.getCaCertificatePath();
+  }
+
+  @Override
+  public CredentialsProvider getCredentialsProvider() {
+    if (!(authentication instanceof DefaultNoopAuthentication)) {
+      return new CredentialsProvider() {
+        @Override
+        public void applyCredentials(Metadata headers) throws IOException {
+          final var authHeader = authentication.getTokenHeader(Product.ZEEBE);
+          final var authHeaderKey = Metadata.Key.of(authHeader.getKey(), Metadata.ASCII_STRING_MARSHALLER);
+          headers.put(authHeaderKey, authHeader.getValue());
+        }
+
+        @Override
+        public boolean shouldRetryRequest(Throwable throwable) {
+          return ((StatusRuntimeException) throwable).getStatus() == Status.DEADLINE_EXCEEDED;
+        }
+      };
+    }
```

Comment:
We need to set a custom auth-url (KeyCloak) for the Zeebe connection which seems not to be implemented yet in 8.3.x what is to reason for introducing this wrapper. I simply tried to reuse what is prepared in `SelfManageAuthentication` since there is the `Product.ZEEBE` enum. I thought this is already prepared to be used in the near future, so I took this right now to get our scenario working. I didn't recognize what's the target picture for the authentication stuff added recently.